### ISSUE TITLE
The spec should be accompanied by examples which are automatically tested

### ISSUE THREAD

**Issue by wilzbach** [rejectedsoftware/diet-ng#30]
I know that this document is incomplete. Hence, this is just an issue to track progress, s.t. it's not forgotten.

The idea is that maybe the spec is generated from a D file with many unittests.

**Comment by quickfur** [comment]
I am having an extremely frustrating time working with Diet templates because the docs are so incomplete.  It wasn't until that lots of time spent googling turned up the Jade Template docs that I found a doc that covers the syntax in a reasonably complete way, albeit with the caveat that Diet isn't 100% compatible with Jade, so half the time it plain doesn't work, or produces a 2-page compile error that's barely understandable.

tl;dr: documented, automatically-tested examples would be highly appreciated.  Also, more complete coverage of the syntax pretty please.

**Comment by wilzbach** [comment]
@quickfur currently the best documentation is the source code itself and the unittest within.

**Comment by quickfur** [comment]
If so, why isn't said unittest part of the doc?!
